### PR TITLE
feat: set JWT token field name with saveToJWT

### PR DESCRIPTION
**Pull request by DanRibbens**
## Description

When using API key authentication with 3rd party systems it can sometimes be necessary to have control over the attribute names in the JWT token.

This change widens the type of `saveToJWT` to be a string that can set the name of the attribute in the JWT token instead of using the field `name`.

- [x] I have read and understand the [CONTRIBUTING.md](../CONTRIBUTING.md) document in this repository.

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## Checklist:

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Existing test suite passes locally with my changes
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: test/auth/config.ts [hunk 1]
```diff
@@ -38,6 +38,12 @@ export default buildConfigWithDefaults({
           saveToJWT: true,
           hasMany: true,
         },
+        {
+          name: 'namedSaveToJWT',
```
Comment (JarrodMFlesch) [inline test/auth/config.ts:42]: I find this example semantically hard to follow, do you think we could adjust the names that we use here? i.e. `fieldName` & `jwtKey` or similar?

**Comment by JarrodMFlesch** [comment]
I did a quick search for `saveToJWT` and I see that it is used in `auth/operations/resetPassword` and `src/collection/graphql`, do those files need adjustment? I feel like reset password will, unsure of the graphql file.

**Comment by DanRibbens** [comment]
Thanks for the review @JarrodMFlesch. I addressed the resetPassword, that was missed. The GraphQL change is not an option unfortunately.

I tried to refactor to allow for the JWT type to support savetToJWT fields inside rows and also the name change feature. The trouble though is that you might might want to have a kebab-case saveToJWT string, but GraphQL doesn't support names that do this.

I wrote this code in `src/collection/graphql/init.ts` that was trying to handle this change (and have to throw it away):
```ts
      const jwtFields = config.fields.reduce((reducedFields, field: Field) => {
        if (!fieldAffectsData(field) && fieldHasSubFields(field)) {
          field.fields.forEach((subField) => {
            if (fieldAffectsData(subField) && subField.saveToJWT) {
              const sanitizedField = subField;
              sanitizedField.name = typeof subField.saveToJWT === 'string' ? subField.saveToJWT : subField.name;
              reducedFields.push(sanitizedField);
            }
          });
        }
        if (fieldAffectsData(field) && field.saveToJWT) {
          const sanitizedField = field;
          sanitizedField.name = typeof field.saveToJWT === 'string' ? field.saveToJWT : field.name;
          reducedFields.push(sanitizedField);
        }
        return reducedFields;
      }, [
        ...authFields,
        {
          name: 'collection',
          type: 'text',
          required: true,
        },
      ]) as FieldAffectingData[];

      collection.graphQL.JWT = buildObjectType({
        payload,
        name: formatName(`${slug}JWT`),
        fields: jwtFields,
        parentName: formatName(`${slug}JWT`),
      });
```

**Comment by JarrodMFlesch** [comment]
@DanRibbens oh interesting, we should document that shortcoming in the docs alongside the saveToJWT definition that you added then.

**Comment by DanRibbens** [comment]
Is it worth it? The actual token will have the same decoded values whether from REST or GraphQL. I don't want there to be any confusion there.

I'm not sure I understand when somebody would use this extra JWT type or if it is even accurate how it is being used in the graphql schema to begin with.
